x-pack/plugins/security_solution/public/timelines/components/side_panel/host_details/index.tsx

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable react/display-name */
+
+import {
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import styled from 'styled-components';
+import { HostDetailsLink } from '../../../../common/components/links';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { useSourcererScope } from '../../../../common/containers/sourcerer';
+import { HostOverview } from '../../../../overview/components/host_overview';
+import { setAbsoluteRangeDatePicker } from '../../../../common/store/inputs/actions';
+import { HostItem } from '../../../../../common/search_strategy';
+import { AnomalyTableProvider } from '../../../../common/components/ml/anomaly/anomaly_table_provider';
+import { hostToCriteria } from '../../../../common/components/ml/criteria/host_to_criteria';
+import { scoreIntervalToDateTime } from '../../../../common/components/ml/score/score_interval_to_datetime';
+import { HostOverviewByNameQuery } from '../../../../hosts/containers/hosts/details';
+
+const StyledEuiFlyoutBody = styled(EuiFlyoutBody)`
+  .euiFlyoutBody__overflow {
+    display: flex;
+    flex: 1;
+    overflow-x: hidden;
+    overflow-y: scroll;
+
+    .euiFlyoutBody__overflowContent {
+      flex: 1;
+      overflow-x: hidden;
+      overflow-y: scroll;
+      padding: ${({ theme }) => `${theme.eui.paddingSizes.xs} ${theme.eui.paddingSizes.m} 64px`};
+    }
+  }
+`;
+
+const StyledEuiFlexGroup = styled(EuiFlexGroup)`
+  flex: 0;
+`;
+
+const StyledEuiFlexItem = styled(EuiFlexItem)`
+  &.euiFlexItem {
+    flex: 1 0 0;
+    overflow-y: scroll;
+    overflow-x: hidden;
+  }
+`;
+
+const StyledEuiFlexButtonWrapper = styled(EuiFlexItem)`
+  align-self: flex-start;
+`;
+
+interface HostDetailsProps {
+  expandedHost: { hostName: string };
+  handleOnHostClosed: () => void;
+  isFlyoutView?: boolean;
+}
+
+export const HostDetailsPanel: React.FC<HostDetailsProps> = React.memo(
+  ({ expandedHost, handleOnHostClosed, isFlyoutView }) => {
+    const { to, from, isInitializing } = useGlobalTime();
+    const { docValueFields, selectedPatterns } = useSourcererScope();
+    const { hostName } = expandedHost;
+
+    if (!hostName) {
+      return null;
+    }
+
+    const ExpandableHostWrapper = () => (
```

Comment:
I think that I completely understand that you create this pattern to have a clean code, but I think it will be better to move these function in the render and let react do its job about memoizing these components. I think we can find a better pattern in the future to avoid that.

Reply:
by the way that's fine, just the function ;)
```
const title = (
      <EuiTitle size="s">
        <h4>
          {i18n.translate('xpack.securitySolution.timeline.sidePanel.hostDetails.title', {
            defaultMessage: 'Host details',
          })}
          {`: ${hostName}`}
        </h4>
      </EuiTitle>
    );
```

Reply:
Yea, but it's extra unnecessary work for the component. I've removed it to it's own file and made it consistent with `event_details`